sdk/identity/azure-identity/src/token_credential_impl.cpp

```diff
@@ -309,13 +337,14 @@ AccessToken TokenCredentialImpl::ParseToken(
       }
     }
 
+    auto const tzOffsetStr = TimeZoneOffsetAsString(utcDiffSeconds);
     if (expiresOn.is_string())
     {
       auto const expiresOnAsString = expiresOn.get<std::string>();
       for (auto const& parse : {
-               std::function<DateTime(std::string const&)>([](auto const& s) {
+               std::function<DateTime(std::string const&)>([&](auto const& s) {
```

Comment:
Why is this change to add `[&]` needed now?

Reply:
That's because we capture `tzOffsetStr` by reference.

Reply:
And just out of curiosity around benefits/drawbacks, why do we capture it by reference?

Reply:
References are less expensive, they are basically pointers, and if we capture by value, we are copying the string - so in case strings are not copyable-on-write (which is very likely), we end up allocating memory and then strcopy.